### PR TITLE
Various code fixes

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
I push a large code change out of this PR, to minimize the review work to do .

### CODE REVIEW

File: crypto/bio/bio_lib.c [hunk 1]
```diff
@@ -45,22 +45,21 @@ int BIO_set(BIO *bio, const BIO_METHOD *method)
     bio->references = 1;
     bio->num_read = 0L;
     bio->num_write = 0L;
-    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_BIO, bio, &bio->ex_data))
-        return 0;
-
     bio->lock = CRYPTO_THREAD_lock_new();
```
Comment (richsalz) [inline crypto/bio/bio_lib.c]: one thing to consider is have a static BIO object with initializers and then do an assignment, to get the initializatoin.  If not that, then just doing memset() and setting the non-zero pointers is better.  while you're in this part of the code.

Comment (FdaSilvaYY) [inline crypto/bio/bio_lib.c]: I don't understand the code-meaning your 1st sentence.
Will reading this code, I also think that the initialisation code should be moved to BIO_new().
Let's do it...

Comment (richsalz) [inline crypto/bio/bio_lib.c]:
```
static BIO nilb = { 0, 0, 0, -1L, 0, 0, …. Etc }

*b = nil;
```

**Comment by kroeckx** [comment]
+1

**Comment by richsalz** [comment]
merged, thanks.

**Pull request closed without merging**